File: backendworker/NHSOnline.Backend.PfsApi/Areas/Configuration/Models/KnownService.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace NHSOnline.Backend.PfsApi.Areas.Configuration.Models
{
    public abstract class KnownService
    {
        [JsonConverter(typeof(StringEnumConverter), false)]
        public JavaScriptInteractionMode JavaScriptInteractionMode { get; set; }
        
        [JsonConverter(typeof(StringEnumConverter), false)]
        public MenuTab MenuTab { get; set; }

        [JsonConverter(typeof(StringEnumConverter), false)]
        public ViewMode ViewMode { get; set; }

        public bool ShowSpinner { get; set; }

        public bool ShowThirdPartyWarning{ get; set; }
        
        public bool RequiresAssertedLoginIdentity { get; set; }
        
        public bool ValidateSession { get; set; }
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/services/knownservices/KnownService.kt
package com.nhs.online.nhsonline.services.knownservices

import com.nhs.online.nhsonline.services.knownservices.enums.JavaScriptInteractionMode
import com.nhs.online.nhsonline.services.knownservices.enums.MenuTab
import com.nhs.online.nhsonline.services.knownservices.enums.ViewMode

interface KnownService {
    var requiresAssertedLoginIdentity: Boolean
    var validateSession: Boolean
    var menuTab: MenuTab
    var viewMode: ViewMode
    var javaScriptInteractionMode: JavaScriptInteractionMode
    var showSpinner: Boolean
}